File: 백준/Silver/1463. 1로 만들기/1로 만들기.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n= sc.nextInt();
        int[] arr = new int[n+1];
        Arrays.fill(arr, 1000001);
        arr[1] = 0;
        for(int i=1; i<n; i++){
            arr[i+1] = Math.min(arr[i+1],arr[i]+1);
            if(i*2 <= n) arr[i*2] = Math.min(arr[i*2],arr[i]+1);
            if(i*3 <= n) arr[i*3] = Math.min(arr[i*3],arr[i]+1);
        }
        System.out.println(arr[n]);
    }
}

File: 백준/Silver/1463. 1로 만들기/1로 만들기.js
let N = Number(require('fs').readFileSync("dev/stdin").toString());
// console.log(N);
const MAX = 1000001;
let memo = new Array(N+1).fill(MAX);

memo[1] = 0;

for(var i=1; i<N; i++) {
  var nowLength = memo[i]+1;
  memo[i+1] = Math.min(memo[i+1], nowLength);
  if(i*2 <= N) memo[i*2] = Math.min(memo[i*2], nowLength);
  if(i*3 <= N) memo[i*3] = Math.min(memo[i*3], nowLength);
}

console.log(memo[N]);
